astropy/logger.py

```diff
@@ -211,7 +211,7 @@ def _showwarning(self, *args, **kwargs):
                     mod_name = mod.__name__
                     break
         else:  # pragma: py2
-            for name, mod in sys.modules.items():
+            for name, mod in list(sys.modules.items()):
```

Comment:
Right - would it make sense to write a ``get_modules`` function that avoids the copy in Python 2?